test/widgets/message_list_test.dart

```diff
@@ -300,4 +300,87 @@ void main() {
       debugNetworkImageHttpClientProvider = null;
     });
   });
+
+  group('_UnreadMarker animations', () {
+    Animation getAnimation(WidgetTester tester, int messageId) {
```

Comment:
This last commit:

30cb3cda5 msglist test [nfc]: Pull out getAnimation helper

has the feeling that within the branch an early commit went one direction, then a later commit decided a different direction was better and went that way instead.

Better to revise the branch as a whole to give it a coherent narrative. That means squashing that commit into the one that originally introduces its own version of `getAnimation`:
8605d5641 msglist: Add animated marker for unread messages

Then this later commit:
765de1648 msglist: Maintain _UnreadMarker animation state when MessageListView.items changes size

has no need to introduce an alternate version, and there's no need for the third commit moving the alternate version to replace the original version.